paddle/majel/malloc.cc

```diff
@@ -0,0 +1,131 @@
+#include "malloc.h"
+#include <glog/logging.h>
+
+#ifndef PADDLE_ONLY_CPU
+#include <cuda_runtime.h>
+#endif
+
+#define CHECK_CUDA(cudaFunc)                                         \
```

Comment:
Let us don't CHECK_EQ here and remove the definition of CHECK_CUDA.

It is not hard to handle the error returned by cudaMalloc. Majel did the following in `src/gpu/detail/cuda.cu`:

```cpp
void* malloc(size_t size) {
    void* ptr = 0;

    cudaError_t result = cudaMalloc(&ptr, size);

    if (result == cudaSuccess) {
        return ptr;
    }

    // clear last error
    cudaGetLastError();

    return nullptr;
}
```